React/Modules/RCTUIManager.m

```diff
@@ -729,9 +729,44 @@ - (void)_amendPendingUIBlocksWithStylePropagationUpdateForShadowView:(RCTShadowV
 
 - (void)_removeChildren:(NSArray<id<RCTComponent>> *)children
           fromContainer:(id<RCTComponent>)container
+              permanent: (BOOL)permanent
 {
+  RCTLayoutAnimation *layoutAnimation = _layoutAnimation;
+  RCTAnimation *deleteAnimation = layoutAnimation.deleteAnimation;
+  
+  __block NSUInteger completionsCalled = 0;
+  
   for (id<RCTComponent> removedChild in children) {
-    [container removeReactSubview:removedChild];
+    
+    void (^completion)(BOOL) = ^(BOOL finished) {
+      completionsCalled++;
+
+      [container removeReactSubview:removedChild];
+      
+      if (layoutAnimation.callback && completionsCalled == children.count) {
+        layoutAnimation.callback(@[@(finished)]);
+        
+        // It's unsafe to call this callback more than once, so we nil it out here
+        // to make sure that doesn't happen.
+        layoutAnimation.callback = nil;
+      }
+    };
+    
+    if (permanent && deleteAnimation && [removedChild isKindOfClass: [UIView class]]) {
+      UIView *view = (UIView *)removedChild;
+      [deleteAnimation performAnimations:^{
+        if ([deleteAnimation.property isEqual:@"scaleXY"]) {
+          view.layer.transform = CATransform3DMakeScale(0, 0, 0);
+        } else if ([deleteAnimation.property isEqual:@"opacity"]) {
+          view.layer.opacity = 0.0;
+        } else {
+          RCTLogError(@"Unsupported layout animation createConfig property %@",
+                      deleteAnimation.property);
+        }
+      } withCompletionBlock:completion];
```

Comment:
I'm worried about weird edge-cases that could happen if the user interacts with the app while this deletion animation is running. Perhaps we should make the view immediately non-interactable? Any other safeguards we could use? How much did you bang on the prototype?